core/src/main/java/org/apache/calcite/interpreter/Bindables.java

```diff
@@ -517,29 +521,50 @@ public Node implement(InterpreterImplementor implementor) {
     }
   }
 
-  /**
-   * Rule to convert an {@link org.apache.calcite.rel.logical.LogicalUnion}
-   * to a {@link BindableUnion}.
-   */
-  public static class BindableUnionRule extends ConverterRule {
+  @Deprecated // use BindableSetOpRule instead
+  public static class BindableUnionRule extends BindableSetOpRule {
 
     /**
      * Creates a BindableUnionRule.
      *
      * @param relBuilderFactory Builder for relational expressions
      */
     public BindableUnionRule(RelBuilderFactory relBuilderFactory) {
```

Comment:
Can we remove this constructor if possible?

Reply:
There is no default constructor available in 'BindableSetOpRule'.

Reply:
ok, thanks